src/audio/AudioPlayer.ts

```diff
@@ -232,13 +246,45 @@ export class AudioPlayer extends EventEmitter {
 			throw new Error(`Cannot play a resource (${resource.name ?? 'unnamed'}) that has already ended.`);
 		}
 
-		this.state = {
-			status: AudioPlayerStatus.Playing,
-			missedFrames: 0,
-			resource,
-			nextTime: Date.now(),
-			stepTimeout: setTimeout(() => this._step(), 0)
-		};
+		if (resource.playStream.readable) {
+			this.state = {
+				status: AudioPlayerStatus.Playing,
+				missedFrames: 0,
+				resource,
+				nextTime: Date.now(),
+				stepTimeout: setTimeout(() => this._step(), 0)
```

Comment:
Is `stepTimeout` supposed to be a timeout and not an immediate? This is equivalent to `setTimeout(() => this._step(), 1)`, as per Node.js's [docs](https://nodejs.org/dist/latest-v14.x/docs/api/timers.html#timers_settimeout_callback_delay_args):

> When `delay` is larger than `2147483647` or less than `1`, the `delay` will be set to `1`. Non-integer delays are truncated to an integer.

Reply:
That's interesting, I didn't know that!

I've adjusted for this by making `stepTimeout` optional (in the very first instance, it will start immediately and so a timeout will not be needed).